xdsl/dialects/riscv.py

```diff
@@ -1531,23 +1531,13 @@ class FMVOp(RdRsOperation[FloatRegisterType, FloatRegisterType]):
     """
     A pseudo instruction to copy contents of one float register to another.
 
-    Equivalent to `fsw rs, 0(sp); flw rd, 0(sp)`.
-
-    This is a much more expensive operation than mv, and does not correspond to an
-    instruction in the RISC-V F extension. Possibly due to this, there is no common
-    assembly pseudo-op for it, so it must be rewritten before assembly printing.
+    Equivalent to `fsw rs, 0(sp); flw rd, 0(sp)` or `fsgnj.s rd, rs, rs`
     """
 
-    name = "riscv.fmv"
+    name = "riscv.fmv.s"
```

Comment:
I know we are not going to handle binary64 for the time being. However, will there be spaces for extension to binary64? Like the class name is `FMVSOp` and `FMVDOp` or something else?

Reply:
It should be pretty easy to add those extensions IMO, or at least I can't see anything stopping us from doing so. Planning-wise, it would probably need to be as a step towards some specific goal.